src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -371,8 +371,17 @@ internal bool CheckValueKind(SyntaxNode node, BoundExpression expr, BindValueKin
                     break;
 
                 case BoundKind.ArrayAccess:
+                    if (RequiresRefAssignableVariable(valueKind))
+                    {
+                        Error(diagnostics, ErrorCode.ERR_RefLocalOrParamExpected, node);
```

Comment:
This is hit through flow analysis. Semantic model is not needed here for the crash.

---
In reply to: [199021151](https://github.com/dotnet/roslyn/pull/28118#discussion_r199021151) [](ancestors = 199021151)

Reply:
The crash was fixed in #27030. This cannot be reproed through semantic model now. My fix is to produce diagnostics instead of emitting invalid IL.

---
In reply to: [199021151](https://github.com/dotnet/roslyn/pull/28118#discussion_r199021151) [](ancestors = 199021151)

Reply:
@OmarTawfik My point is that we must have never exercised this code path through the semantic model. I think we should have a regression test, even if it passes with flying colors on the current code.